projects/wvr-elements/src/lib/core/manifest/manifest.reducers.spec.ts

```diff
@@ -0,0 +1,330 @@
+import { EntityMapOne, Update } from '@ngrx/entity';
+import { Manifest } from './manifest';
+import { ManifestEntry } from './manifest-entry';
+import * as fromManifestReducers from './manifest.reducers';
+import * as fromManifestActions from './manifest.actions';
+import { ManifestEntryRequest } from './manifest-entry-request';
+
+describe('Manifest Reducer', () => {
+
+  // START
+
+  const manifestEntry1: ManifestEntry = {
+    name: "All Sorted",
+    path: "/ldap/all-sorted",
+    methods: ["GET"]
+  };
+  const manifestEntry2: ManifestEntry = {
+    name: "Departments",
+    path: "/ldap/departments",
+    methods: ["GET"]
+  };
+
+  const manifest: Manifest = {
+    name: "Directory App",
+    baseUrl: "https://api-dev.library.tamu.edu/directory-service",
+    entries: [manifestEntry1, manifestEntry2]
+  };
+
+  describe('initial state', ()=> {
+    it('should return the initial state', () => {
+      const { initialState } = fromManifestReducers;
+      const action = {} as any;
+      expect(fromManifestReducers.reducer(undefined, action))
+      .toBe(initialState);
+    });
+
+    it('should select manifest by name', () => {
+      expect(fromManifestReducers.selectManifestByName(manifest))
+      .toEqual(manifest.name);
+    });
+    const state =  {currentRequest: undefined,entities: {},ids: [],pendingRequests: []};
+    it('should select current request', () => {
+      expect(fromManifestReducers.selectCurrentRequest(state))
+      .toEqual(state.currentRequest);
+    });
+
+    it('should select pending request', () => {
+      expect(fromManifestReducers.selectPendingRequests(state))
+      .toEqual(state.pendingRequests);
+    });
+
+  });
+
+  describe('Manifest reducer ', () => {
+    const state = { ids: [], entities: {}, pendingRequests: [{manifestName: "Directory App", entryName: "All Sorted"}], currentRequest: undefined };
+    const addManifestActionObj = { manifest: { manifest,entries: [
+            { description: undefined, map: data => data.payload[Object.keys(data.payload)[0]], methods: ["GET"], name: "All Sorted", options: {}, path: "/ldap/all-sorted"},
+            { description: undefined, map: data => data, methods: ["GET"], name: "Departments", options: {}, path: "/ldap/departments" }],
+        name: "Directory App"},
+     type: "[Manifest] Add Manifest"
+    };
+
+    it("should have manifest action type as '[Manifest] Add Manifest'", () => {
+      const addManifestAction = fromManifestActions.addManifest({manifest});
+      expect(JSON.stringify(addManifestAction.type) === JSON.stringify('[Manifest] Add Manifest'))
```

Comment:
Since `addManifestAction.type` is of type `string` I do not believe JSON.stringify is needed.

Reply:
JSON stringify not needed - removed it.

Reply:
removed stringify()